### PR TITLE
Bump ui-box@4.0.0 & dogfood ref support throughout Evergreen

### PR DESCRIPTION
**Pull request by akleiner2**
## Overview 
Now that `ui-box` officially uses `forwardRef` to enable `ref` support to underlying `<Box />` components, this PR upgrade `ui-box` to next major, and updates any references to `<Box innerRef={...} />` to `<Box ref={...} />`. In addition, it updates several callsites to directly pass `ref` through instead of `innerRef`,  especially now that a vast majority of Evergreen has `forwardRef`.
 
## Screenshots (if applicable) 
N/A -- looked through Storybook extensively, and component behavior seems intact. 

## Testing
- [ ] Updated Typescript types and/or component PropTypes 
- [x] Added / modified component docs 
- [ ] Added / modified Storybook stories

### CODE REVIEW

File: src/tooltip/src/Tooltip.js [hunk 1]
```diff
@@ -209,7 +209,7 @@ export default class Tooltip extends PureComponent {
           <TooltipStateless
             id={this.state.id}
             appearance={appearance}
-            innerRef={ref => getRef(ref)}
+            ref={ref => getRef(ref)}
```
Comment (mshwery) [inline src/tooltip/src/Tooltip.js:212]: Seems like this could just be `ref={getRef}`

File: src/tooltip/src/Tooltip.js [hunk 2]
```diff
@@ -155,7 +155,7 @@ export default class Tooltip extends PureComponent {
      */
     return React.cloneElement(children, {
       ...tooltipTargetProps,
-      innerRef: ref => {
+      ref: ref => {
```
Comment (mshwery) [inline src/tooltip/src/Tooltip.js:158]: Same here: `ref: getRef`?

**Comment by akleiner2** [comment]
@mshwery yep, that's the idea!

**Comment by mshwery** [comment]
Any update here @akleiner2 ?

**Comment by akleiner2** [comment]
@mshwery Need to rebase on latest v5. @colinlohner will likely need to rebase his table PR too. Will push through today! thanks for the nudeg

**Pull request merged**